test/tests/functional/pbs_snapshot_unittest.py

```diff
@@ -300,3 +302,62 @@ def test_snapshot_without_logs(self):
         # Cleanup
         self.du.rm(path=snap_dir, recursive=True, force=True)
         self.du.rm(path=output_tar, sudo=True, force=True)
+
+    def test_obfuscate_resv_user_groups(self):
+        """
+        Test obfuscation of user & group related attributes while capturing
+        snapshots via pbs_snapshot
+        """
+        now = int(time.time())
+
+        # Let's submit a reservation with Authorized_Users and
+        # Authorized_Groups set
+        attribs = {ATTR_auth_u: TEST_USER1, ATTR_auth_g: TSTGRP0,
+                   ATTR_l + ".ncpus": 2, 'reserve_start': now + 25,
+                   'reserve_end': now + 45}
+        resv_obj = Reservation(attrs=attribs)
+        resv_id = self.server.submit(resv_obj)
+        attribs = {'reserve_state': (MATCH_RE, 'RESV_CONFIRMED|2')}
+        self.server.expect(RESV, attribs, id=resv_id)
+
+        # Now, take a snapshot with --obfuscate
+        target_dir = os.getcwd()
+        snap_cmd = ["pbs_snapshot", "-o", target_dir, "--daemon-logs=0",
+                    "--accounting-logs=0", "--obfuscate"]
+        ret = self.du.run_cmd(cmd=snap_cmd, sudo=True)
+        self.assertEquals(ret['rc'], 0)
+
+        # Get the name of the tarball that was created
+        # pbs_snapshot prints to stdout only the following:
+        #     "Snapshot available at: <path to tarball>"
+        self.assertTrue(len(ret['out']) > 0)
+        snap_out = ret['out'][0]
+        output_tar = snap_out.split(":")[1]
+        output_tar = output_tar.strip()
+
+        # Check that the output tarball was created
+        self.assertTrue(os.path.isfile(output_tar))
+
+        # Unwrap the tarball
+        tar = tarfile.open(output_tar)
+        tar.extractall()
+        tar.close()
+
+        # snapshot directory name = <snapshot>.tgz[:-4]
+        snap_dir = output_tar[:-4]
+
+        # Check that the directory exists
+        self.assertTrue(os.path.isdir(snap_dir))
+
+        # Make sure that the pbs_rstat -f output captured doesn't have the
+        # Authorized user and group names
+        pbsrstat_path = os.path.join(snap_dir, PBS_RSTAT_F_PATH)
+        self.assertTrue(os.path.isfile(pbsrstat_path))
+        with open(pbsrstat_path, "r") as rstatfd:
```

Comment:
Overall change looks good. I have a nit about what happens to the test case if pbsrstat_path fails to open. I think you may way to fail the test in that case.

I also have a general question... when you execute snapshot as a super user, what are the permissions of the tarball created in target directory? If I run this PTL script as a normal user will it be able to read the snapshot? If not then we should make sure that the script is always run as root user. what do you think?